### PR TITLE
Rapier transition

### PR DESCRIPTION
**Pull request by OwenBoisse**
changed physics system from bevy xpbd to bevy rapier

### CODE REVIEW

File: src/spells.rs [hunk 1]
```diff
@@ -479,57 +514,69 @@ fn handle_parry(
...
-                    RigidBody::Dynamic,
-                ))
-                .add_rollback();
-            // only parry one thing per frame - quick n dirty way to not spawn two parried guys
-            // if we hit one normal projectile with two hands
-            return;
+    for (parry_obj, parry_transform, _) in parry_objs_query.iter_mut() {
+        for collision in &collisions {
+            match collision {
+                CollisionEvent::Started(entity1, entity2, _) => {
```
Comment (rosefromthedead) [inline src/spells.rs:532]: `let CollisionEvent::Started(entity1, entity2, _) = collision else { continue };`

File: src/spells.rs [hunk 2]
```diff
@@ -331,22 +341,36 @@ fn hand_bomb_collision(
...
+                        if (*entity1 == hand_entity && *entity2 == bomb_entity)
+                            || (*entity2 == hand_entity && *entity1 == bomb_entity)
+                        {
+                            let hand_bomb_direction =
+                                (bomb_trans.translation - hand_transform.translation).normalize();
+
+                            commands.entity(bomb_entity).insert(RigidBody::Dynamic);
+                            commands
+                                .entity(bomb_entity)
+                                .insert(ExternalImpulse::at_point(
```
Comment (rosefromthedead) [inline src/spells.rs:360]:
```suggestion
                            commands
                                .entity(bomb_entity)
                                .insert(RigidBody::Dynamic)
                                .insert(ExternalImpulse::at_point(
```

**Review by rosefromthedead — APPROVED** [review]
A couple of minor things that I'd appreciate being polished, but not important enough to block

**Review by rosefromthedead — APPROVED** [review]
Ty

**Review by Foxinatel — APPROVED** [review]
LGTM

**Pull request merged**